roles/confluent.schema_registry/tasks/main.yml

```diff
@@ -39,6 +43,15 @@
     system: true
     group: "{{schema_registry_group}}"
 
+# Archive File deployments need to create SystemD service units
+- name: Create Schema Registry Service for archive file deployments
```

Comment:
This is a little tedious, but the systemd service files are included in the tar. And our preferred method of updating the service files is with the override.conf files. 

I'd like this to be a `copy` task, and then look at what is templatized and figure out how to put that into the overrides. see `schema_registry_service_overrides` dict within `roles/confluent.schema_registry/defaults/main.yml`

The reason I want to use the service file from the tar is because if the service file changes in the future, then cp-ansible will not need to change

This goes for each components service file.

Reply:
I honestly didn't even know those were included in the tarball. Looking at them it could be a copy task and then a few replace tasks with regexes to put in the correct paths, username and group. Thoughts?

Reply:
Yes copy and then line replacement would work. But the standard we already have is
- get systemd service file (with yum)
- anything that needs to be overwritten gets added to the service override.conf file. Notice we already have logic for the user/group: https://github.com/confluentinc/cp-ansible/blob/6.0.x/roles/confluent.schema_registry/defaults/main.yml#L19

It seems the other logic your template contains is the `ExecStart` line which I think could be added to the dict like this

```
schema_registry_service_overrides:
  LimitNOFILE: 100000
  User: "{{ schema_registry_user if schema_registry_user != schema_registry_default_user else '' }}"
  Group: "{{ schema_registry_group if schema_registry_group != schema_registry_default_group else '' }}"
  ExecStart="{% if archive_file_enabled |bool %}{{ schema_registry.server_start_file }} {{ schema_registry.config_file }}{% endif %}"
```

Reply:
Even easier and I learned something new. I'll get those changes made.

Reply:
Fixed in commit 263b8b5a. The service files are copied from the tarball into `/usr/lib/systemd/system` and the overrides contain an ExecStart to the tarballs binary. Note there is a blank `ExecStart=` because systemd needs to clear the ExecStart in an override before re-assigning it.